### DISCUSSION TITLE
Confirmation modal on tab change

### DISCUSSION THREAD

**Discussion by Joshi-Kajal** [twilio-labs/paste#2116] [Q&A] [answered]
Hey team, we have one use case where we want to have confirmation before tab change about discarding any unsaved user changes.

Flow we expect: 
- click on tab
- If there is unsaved changes, ask for confirmation to leave tab
- If user click on confirm, move to next selected tab
- If user click on cancel, stay on same tab

[useTabState](https://paste.twilio.design/primitives/tabs-primitive/#paste-docs-content-area) does not give much flexibility to listen and prevent tab change.

Please guide me here how to achieve this behaviour.

**Comment by TheSisb** [comment]
Hi @Joshi-Kajal, apologies for the slow response time. I'll be looking into this and getting back to you shortly.

**Comment by TheSisb** [comment, marked as answer]
Hi @Joshi-Kajal,

I've managed to find [a working solution](https://codesandbox.io/s/wispy-sound-v7xv1?file=/src/index.tsx) by overwriting the `select` function generated by the `useTabState` hook. Let me know if you have any other questions or concerns with this approach.
